simulator/autoware_auto_msgs_adapter/include/autoware_auto_msgs_adapter/adapter_base_interface.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 The Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef AUTOWARE_AUTO_MSGS_ADAPTER__ADAPTER_BASE_INTERFACE_HPP_
+#define AUTOWARE_AUTO_MSGS_ADAPTER__ADAPTER_BASE_INTERFACE_HPP_
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <memory>
+
+namespace autoware_auto_msgs_adapter
+{
+
+class AdapterBaseInterface
```

Comment:
My plan was to be able to initialize the same node as different adapters based on the `msg_type_target`.

An instance can become a selected adapter.

(One node will only have one adapter based on the selection on launch.)

Like:
```cpp
  if (msg_type_target == "autoware_auto_control_msgs::msg::AckermannControlCommand") {
    AdapterControl::SharedPtr adapter =
      std::make_shared<AdapterControl>(*this, topic_name_source, topic_name_target);
    adapter_ = std::dynamic_pointer_cast<AdapterBaseInterface>(adapter);
  } else if (msg_type_target == "autoware_auto_planning_msgs::msg::Message2") {
    AdapterPlanning::SharedPtr adapter =
      std::make_shared<AdapterPlanning>(*this, topic_name_source, topic_name_target);
    adapter_ = std::dynamic_pointer_cast<AdapterBaseInterface>(adapter);
  } else {
    RCLCPP_ERROR(get_logger(), "Unknown msg type: %s", msg_type_target.c_str());
  }
```

I don't understand if you have a suggestion on this. Because:
- I cannot turn it into a vector, I would still need the Interface class to be able to make a vector of same type. But I need only 1 adapter per node, this is not useful.
- I cannot replace `adapter_` with `AdapterControl` type because I want to be able to transform the node into different types.